client/server-config-storage.ts

```diff
@@ -0,0 +1,6 @@
+import { ServerConfig } from '../common/server-config'
+import { JsonLocalStorageValue } from './local-storage'
+import { getServerOrigin } from './network/server-url'
+
+const serverOrigin = getServerOrigin().toLowerCase()
```

Comment:
So this I guess is the problematic call? I would just pull the `baseUrl` thing into its own file and import that in both files, and use that as the key for localStorage (in Electron, in the browser you don't really need a unique key for this at all since each origin will get its own localStorage, so an empty string or any arbitrary string is fine there)